source/Calamari.Common/Plumbing/FileSystem/CalamariPhysicalFileSystem.cs

```diff
@@ -389,19 +385,22 @@ string GetTempBasePath()
             path1 = Path.Combine(path1, Assembly.GetEntryAssembly()?.GetName().Name ?? "Octopus.Calamari");
             path1 = Path.Combine(path1, "Temp");
             var path = path1;
-            if (!Directory.Exists(path))
-                Directory.CreateDirectory(path);
+            Directory.CreateDirectory(path);
             return Path.GetFullPath(path);
         }
 
         public string CreateTemporaryDirectory()
         {
             var path = Path.Combine(GetTempBasePath(), Guid.NewGuid().ToString());
-            if (!Directory.Exists(path))
-                Directory.CreateDirectory(path);
+            CreateDirectory(path);
             return path;
         }
 
+        public void CreateDirectory(string directory)
+        {
+            Directory.CreateDirectory(directory);
+        }
```

Comment:
It's unfortunately required because Dictionary is a property not just the static class. It can contain different implementations depending on the calamari build - for .net Framework it uses `LongPathsDirectory` for everything else it uses `StandardDictionary`.

Reply:
Oh right, I missed that this is a public method.

Reply:
Also, Dictionary => Directory for a bunch of your comments here 😁

Reply:
woops 🤦‍♂️